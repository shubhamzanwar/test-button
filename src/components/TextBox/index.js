import React from 'react'

function TextBox({testId, onChange, value}) {
    return (
        <div>
            <input type="text" value={value} data-testid={testId} onChange={(e)=>onChange(e.target.value)}></input>
        </div>
    )
}

export default TextBox
