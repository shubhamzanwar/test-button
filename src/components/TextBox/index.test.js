import React from 'react';
import TextBox from './index';
import { render,fireEvent } from '@testing-library/react';

describe('The TextBox Component', () => {
    it('should render correctly', () => {
        const { asFragment } = render(<TextBox testID="test-input" type="text"/>);

        expect(asFragment()).toMatchSnapshot();
    });
    it('should call the onChange prop when there is a change in the input box', () => {
        const onChangeMock = jest.fn();
        const { getByTestId } = render(<TextBox testID="test-input" type="text" onChange={onChangeMock} />);

        fireEvent.change(getByTestId("test-input"), {target: {value:'sarvani'}});
        expect(onChangeMock).toHaveBeenCalledWith('sarvani');
    });
});