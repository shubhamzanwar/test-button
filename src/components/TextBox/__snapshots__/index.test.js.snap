// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The TextBox component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <input
      data-testid="123"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
